File: lib/duplicateCount.dart
int duplicateCount(String text) {

  var array = text.toLowerCase().split('');

  print(array);

  Map<String, int> map = Map.fromIterable(array, key: (item) => item, value: (item) => 0);

	map.forEach((String elementMap, int key) {
		array.forEach((String elementArray) {
			if (elementArray.toLowerCase() == elementMap.toLowerCase()) map[elementMap]++;
		});
  });

	var occurs = 0;

	map.forEach((key, value) {
		if (value > 1) occurs++;
	});

	print(map.length);
	print(array.length);

  return occurs;
}

File: js/duplicateCount.ts
function duplicateCount(text: string): number {

	let array: string[] = text.toLowerCase().split('');
	let answer: string[] = [];

	console.log(array);

	for (let i = 0; i < array.length; i++) {
		let letter = array.shift();
		console.log(letter);
		if (array.includes(letter)) {
			if (!answer.includes(letter)) answer.push(letter);
		}
	}

	console.log(answer.length);
	return answer.length;
}

duplicateCount('aabBcde');